### PR TITLE
fix(TDI-46665): support java 17

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-46665

**What is the new behavior?**
https://jira.talendforge.org/browse/TDI-46665

**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/951/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/951/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/952/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/952/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/953/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/953/)

File: components-parent/components-bom/pom.xml [hunk 1]
```diff
@@ -1313,6 +1314,11 @@
                 <artifactId>daikon</artifactId>
                 <version>${daikon.version}</version>
             </dependency>
+            <dependency>
```
Comment (wwang-talend) [inline components-parent/components-bom/pom.xml:1317]: add it as avoid to upgrade daikon...

**Comment by wwang-talend** [comment]
https://jenkins-connectors.datapwn.com/job/components/job/build/job/wwang-talend%252FTDI-46665/ pass junit

**Review by pyzhou-talend — APPROVED** [review]
LGTM

**Pull request merged**